### PR TITLE
Remove `AggregateFunctionDefinition`

### PR DESCRIPTION
**Pull request by lewiszlw**
## Which issue does this PR close?

Closes #.

## Rationale for this change

Remove `AggregateFunctionDefinition` as all aggr functions are already implemented using UDAF.

### CODE REVIEW

File: datafusion/core/src/physical_planner.rs [hunk 1]
```diff
@@ -223,7 +223,7 @@ fn create_physical_name(e: &Expr, is_first_expr: bool) -> Result<String> {
             create_function_physical_name(&fun.to_string(), false, args, Some(order_by))
         }
         Expr::AggregateFunction(AggregateFunction {
-            func_def,
+            func: func_def,
```
Comment (alamb) [inline datafusion/core/src/physical_planner.rs:226]: nit is that we could rename the local variable to func as well for more consistency

File: datafusion/expr/src/expr.rs [hunk 2]
```diff
@@ -627,22 +627,6 @@ impl Sort {
     }
 }
 
-#[derive(Debug, Clone, PartialEq, Eq, Hash)]
```
Comment (alamb) [inline datafusion/expr/src/expr.rs:630]: 🎉 🧹

File: datafusion/expr/src/expr.rs [hunk 3]
```diff
@@ -674,7 +658,7 @@ impl AggregateFunction {
         null_treatment: Option<NullTreatment>,
     ) -> Self {
         Self {
-            func_def: AggregateFunctionDefinition::UDF(udf),
+            func: udf,
```
Comment (alamb) [inline datafusion/expr/src/expr.rs:661]: ditto here it could be even more consistent if all the names were the same

**Review by alamb — APPROVED** [review]
Thank you @lewiszlw  -- this looks very nice to me

cc @jayzhan211

**Review by jayzhan211 — APPROVED** [review]
👍  
Thanks @lewiszlw @alamb

**Pull request merged**